internal/cloud/metadata/metadata.go

```diff
@@ -30,6 +30,8 @@ type InstanceMetadata struct {
 	// AliasIPRanges is a list of IP ranges that are attached.
 	// May be empty on certain CSPs.
 	AliasIPRanges []string
+	// InitSecretHash is the hash of the init secret.
+	InitSecretHash string
```

Comment:
I would prefer to have a function to retrieve `InitSecretHash`, instead of having it as part of `InstanceMetadata`.
The current code already retrieves all the hash information in separate functions to the main `Self` method, so there shouldn't be much to refactor